Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageCollectionViewCell.swift

```diff
@@ -74,3 +74,54 @@ public final class _ChatMessageCollectionViewCell<ExtraData: ExtraDataTypes>: _C
         return preferredAttributes
     }
 }
+
+/// The cell that displays the message content of a dynamic type and layout.
+/// Once the cell is set up it is expected to be dequeued for messages with
+/// the same content and layout the cell has already been configured with.
+public final class _ChatMessageTableViewCell<ExtraData: ExtraDataTypes>: _TableViewCell {
+    public static var reuseId: String { "\(self)" }
+
+    public private(set) var messageContentView: _ChatMessageContentView<ExtraData>?
+    
+    override public func setUp() {
+        super.setUp()
+        
+        selectionStyle = .none
+        backgroundColor = .clear
+        backgroundView = nil
+    }
+
+    override public func prepareForReuse() {
+        super.prepareForReuse()
+
+        messageContentView?.prepareForReuse()
+    }
+
+    public func setMessageContentIfNeeded(
+        contentViewClass: _ChatMessageContentView<ExtraData>.Type,
+        attachmentViewInjectorType: _AttachmentViewInjector<ExtraData>.Type?,
+        options: ChatMessageLayoutOptions
+    ) {
+        guard messageContentView == nil else {
+            log.assert(type(of: messageContentView!) == contentViewClass, """
+            Attempt to setup different content class: ("\(contentViewClass)").
+            `СhatMessageCollectionViewCell` is supposed to be configured only once.
+            """)
+            return
+        }
+
+        messageContentView = contentViewClass.init().withoutAutoresizingMaskConstraints
+        // We add the content view to the view hierarchy before invoking `setUpLayoutIfNeeded`
+        // (where the subviews are instantiated and configured) to use `components` and `appearance`
+        // taken from the responder chain.
+        contentView.addSubview(messageContentView!)
+        messageContentView?.pin(anchors: [.leading, .top, .trailing, .bottom], to: contentView)
+        messageContentView!.setUpLayoutIfNeeded(options: options, attachmentViewInjectorType: attachmentViewInjectorType)
+        
+        if options.contains(.continuousBubble) && !options.contains(.threadInfo) {
+            messageContentView?.mainContainer.layoutMargins.bottom = 2
+        } else {
+            messageContentView?.mainContainer.layoutMargins.bottom = 8
+        }
```

Comment:
I'm still not sure about this. Why exactly did you move this code here?

Reply:
had a call with Vojta regarding the spacing, came up with following adjustments 7f1803a924547bbead2b310e7e47e13ded16f528